docs/rst/use_cases/keys/keys.rst

```diff
@@ -0,0 +1,171 @@
+The Qt application terminal displays the turtles' positions, and the controller terminal shows the commands sent to each Turtle Node.
+The controller receives the turtles position but it has subscriber filtered to show only the positions of turtles 1 and 2.
+Each velocity and pose message is tagged with a key representing the turtle ID.
+
+The following diagram illustrates the system architecture:
+
+.. figure:: /rst/figures/use_cases/keys/keys.png
+    :align: center
+
+.. note::
+
+    Topic keys work exclusively with publishers and subscribers, not with services.
+    Multiple service clients can use the same service,
+    but there can only be one service server for a given service.
+    Since each Turtle Node must have a service server to control goals,
+    it does not make sense to have multiple clients (e.g., controllers) using the same service server with different keys.
+
+
+Prerequisites
+-------------
+
+Familiarity with the following topics is recommended:
+
+* `Content Filter Topic <https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/topic/contentFilteredTopic/contentFilteredTopic.html>`__
+* :ref:`topic_keys`
+
+It is required to have previously installed Vulcanexus using one of the following installation methods:
+
+* :ref:`linux_binary_installation`
+* :ref:`linux_source_installation`
+* :ref:`docker_installation`
+
+
+Keyed Messages
+--------------
+
+The keyed Topics use the following message types to represent the turtles' velocity and pose:
+
+.. raw:: html
+
+    <div style="display: flex;">
+        <div style="flex: 1; padding: 10px;">
+            <p><b>docs_turtlesim::msg::KeyedPose</b></p>
+            <p>Represents the pose of a turtle with a key.</p>
+            <ul>
+                <li>key: Identifier of the turtle.</li>
+                <li>x: X-coordinate of the turtle's position.</li>
+                <li>y: Y-coordinate of the turtle's position.</li>
+                <li>theta: Orientation angle of the turtle.</li>
+            </ul>
+        </div>
+        <div style="flex: 1; padding: 10px;">
+            <img src="../../../_static/resources/use_cases/keys/keyed_pose.png" alt="KeyedPose" style="width: 220px;">
+        </div>
+    </div>
+
+.. raw:: html
+
+    <div style="display: flex;">
+        <div style="flex: 1; padding: 10px;">
+            <p><b>docs_turtlesim::msg::KeyedVelocity</b></p>
+            <p>Represents the velocity of a turtle with a key.</p>
+            <ul>
+                <li>key: Identifier of the turtle.</li>
+                <li>linear: Linear velocity of the turtle.</li>
+                <li>angular: Angular velocity of the turtle.</li>
+            </ul>
+        </div>
+        <div style="flex: 1; padding: 10px;">
+            <img src="../../../_static/resources/use_cases/keys/keyed_twist.png" alt="KeyedTwist" style="width: 320px;">
+        </div>
+    </div>
```

Comment:
I do not think we need this. Remove the images and use just IDL code right after the member explanation. Maybe you can only explain the keyed members